iroh-memesync/src/message.rs

```diff
@@ -0,0 +1,222 @@
+use std::fmt::{Debug, Display};
+
+use bytes::{Bytes, BytesMut};
+use cid::Cid;
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+
+use crate::Error;
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub enum Message {
+    Response(Response),
+    Request(Request),
+}
+
+impl Message {
+    /// Decoding from a given `BytesMut`.
+    pub fn from_bytes(bytes: BytesMut) -> Result<Self, Error> {
+        let msg = bincode::deserialize(&bytes)?;
```

Comment:
CBOR? Just for cross language compatibility maybe?

Reply:
I am still very undecided on the format, not a big fan of cbor, so for now this is a placeholder encoding